Content.Server/_NF/Holiday/HalloweenSystem.cs

```diff
@@ -0,0 +1,39 @@
+using System.Linq;
+using Robust.Shared.Map.Events;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Holiday;
+
+public sealed class HalloweenSystem : EntitySystem
+{
+    [ValidatePrototypeId<HolidayPrototype>]
+    private const string HolidayId = "Halloween";
+
+    [ValidatePrototypeId<EntityPrototype>]
+    private const string DeepFryerId = "KitchenDeepFryer";
+    [ValidatePrototypeId<EntityPrototype>]
+    private const string DeepFryerPOIId = "KitchenDeepFryerPOI";
+
+    [ValidatePrototypeId<EntityPrototype>]
+    private const string CauldronId = "KitchenDeepFryerCauldron";
+    [ValidatePrototypeId<EntityPrototype>]
+    private const string CauldronPOIId = "KitchenDeepFryerCauldronPOI";
```

Comment:
This is not good, hardcoding protoIds inside a system is a bad practice , aspecially because this becomes a larger list over time
need to use `HolidayVisualsComponent` and use `Holidays` var see nuke.yml
is this possible instead ?

Reply:
> is this possible instead ?

No. The cauldron also has a unique name and description, which `HolidayVisualsComponent` can't accommodate. I can't find anything that takes care of "replace entities with other entities on certain holidays", and I don't see the need to build it into `HolidaySystem` as a generic feature until we need it generalised. Premature generalisation is the death of software, and SS14 has far too much of that already.

Reply:
I think you know why I made these comments and it would not suprise me if you found the implementation unnerving, i see you want the entire entity replaced.. but given these circumstances i guess this is the only way. 
Perhaps we can list the proto ids in a yaml instead ? but that probably would not validate them .. GRRRR

Reply:
No point in making more components like HolidayRsiSwapComponent either thats just dumb in a way

Reply:
This _isn't_ just an RSI swap. That's why it's its own system. If it was literally just a sprite change, HolidayVisuals would solve it with absolutely no downsides. :>

Reply:
yeah, i quickly realized after looking more into it

Reply:
For the record, I'm happy to PR for the removal of the `HalloweenSystem` after Halloween and we can develop a more robust and scalable solution, if necessary, and maybe even leave the cauldron entity as a fun extra plaything for admins.

Reply:
When i think about it **this halloween system should legitimately replace the holiday system** in my opinion because it utilizes the entity system better, and does not overcomplicate the sprite layers with holiday cases, instead, people should just make a new entity and set the original one as the parent, and override the sprite system. 
BOOM

Reply:
There *is* a generic `IHolidayCelebrate` interface that you can implement and attach to a specific holiday, and its `Celebrate()` method will be called before the round starts. Unfortunately it's a bit limited in its capabilities and runs too late. :\